src/inmanta/module.py

```diff
@@ -1985,6 +1985,9 @@ def __init__(
         else:
             self.strict_deps_check = self._metadata.strict_deps_check
 
+        self._complete_ast: Optional[Tuple[List[Statement], List[BasicBlock]]]= None
```

Comment:
@sanderr : is this safe?

Reply:
I am relatively confident now that it's a bug that could affect pytest-inmanta.

Reply:
Wow, I wasn't even aware we could unload. 
Do we ever use this in fact?

Reply:
We do. The prime use case is [pytest-inmanta](https://github.com/inmanta/pytest-inmanta/blob/7e78007acede7cf3a21b1d78e82675dffa9f8fd3/pytest_inmanta/plugin.py#L601). We introduced that to make pytest-inmanta compatible with modules v2. I don't recall all details but there were issues with the compiler or even Python env caching things that shouldn't be cached ([test case](https://github.com/inmanta/pytest-inmanta/blob/master/examples/testhandler/tests/test_dynamic_modules.py) for context).

We also use it internally in core in a few places.

Reply:
The test case I linked only seems to test the Python part. I think that changes to the model files would be broken due to the ast cache.

Reply:
`Module.unload` is what `pytest-inmanta` uses.

Reply:
Side note: we should probably use the `stable_api` method on methods as well as classes.